nipype/interfaces/fsl/tests/test_maths.py

```diff
@@ -84,377 +93,330 @@ def test_maths_base(fsl_output_type=None):
     thresh.inputs.use_robust_range = True
-    yield assert_equal, thresh.cmdline, cmdline % ("-uthrp " + val)
+    assert thresh.cmdline == cmdline % ("-uthrp " + val)
     thresh.inputs.use_nonzero_voxels = True
-    yield assert_equal, thresh.cmdline, cmdline % ("-uthrP " + val)
+    assert thresh.cmdline == cmdline % ("-uthrP " + val)
 
-    # Clean up our mess
-    clean_directory(testdir, origdir)
-    set_output_type(prev_type)
 
-
-@skipif(no_fsl)
-def test_meanimage(fsl_output_type=None):
-    prev_type = set_output_type(fsl_output_type)
-    files, testdir, origdir, out_ext = create_files_in_directory()
+@pytest.mark.skipif(no_fsl(), reason="fsl is not installed")
+def test_meanimage(create_files_in_directory):
+    files, testdir, out_ext = create_files_in_directory
 
     # Get the command
     meaner = fsl.MeanImage(in_file="a.nii", out_file="b.nii")
 
     # Test the underlying command
-    yield assert_equal, meaner.cmd, "fslmaths"
+    assert meaner.cmd == "fslmaths"
 
     # Test the defualt opstring
-    yield assert_equal, meaner.cmdline, "fslmaths a.nii -Tmean b.nii"
+    assert meaner.cmdline == "fslmaths a.nii -Tmean b.nii"
 
     # Test the other dimensions
     cmdline = "fslmaths a.nii -%smean b.nii"
     for dim in ["X", "Y", "Z", "T"]:
         meaner.inputs.dimension = dim
-        yield assert_equal, meaner.cmdline, cmdline % dim
+        assert meaner.cmdline == cmdline % dim
 
     # Test the auto naming
     meaner = fsl.MeanImage(in_file="a.nii")
-    yield assert_equal, meaner.cmdline, "fslmaths a.nii -Tmean %s" % os.path.join(testdir, "a_mean%s" % out_ext)
-
-    # Clean up our mess
-    clean_directory(testdir, origdir)
-    set_output_type(prev_type)
+    assert meaner.cmdline == "fslmaths a.nii -Tmean %s" % os.path.join(testdir, "a_mean%s" % out_ext)
 
-@skipif(no_fsl)
-def test_stdimage(fsl_output_type=None):
-    prev_type = set_output_type(fsl_output_type)
-    files, testdir, origdir, out_ext = create_files_in_directory()
+    
+@pytest.mark.skipif(no_fsl(), reason="fsl is not installed")
+def test_stdimage(create_files_in_directory):
+    files, testdir, out_ext = create_files_in_directory
 
     # Get the command
     stder = fsl.StdImage(in_file="a.nii",out_file="b.nii")
 
     # Test the underlying command
-    yield assert_equal, stder.cmd, "fslmaths"
+    assert stder.cmd == "fslmaths"
 
     # Test the defualt opstring
-    yield assert_equal, stder.cmdline, "fslmaths a.nii -Tstd b.nii"
+    assert stder.cmdline == "fslmaths a.nii -Tstd b.nii"
 
     # Test the other dimensions
     cmdline = "fslmaths a.nii -%sstd b.nii"
     for dim in ["X","Y","Z","T"]:
         stder.inputs.dimension=dim
-        yield assert_equal, stder.cmdline, cmdline%dim
+        assert stder.cmdline == cmdline%dim
 
     # Test the auto naming
     stder = fsl.StdImage(in_file="a.nii")
-    yield assert_equal, stder.cmdline, "fslmaths a.nii -Tstd %s"%os.path.join(testdir, "a_std.nii")
```

Comment:
ok, I've changed to `stder = fsl.StdImage(in_file="a.nii", output_type='NIFTI')` and test doesn't fail anymore